share/new_eds/eds.go

```diff
@@ -10,16 +10,17 @@ import (
 	"github.com/celestiaorg/celestia-node/share/shwap"
 )
 
-type EdsFile interface {
+// EDS is an interface for accessing extended data square data.
+type EDS interface {
 	io.Closer
 	// Size returns square size of the file.
 	Size() int
-	// Share returns share and corresponding proof for the given axis and share index in this axis.
-	Share(ctx context.Context, rowIdx, colIdx int) (*shwap.Sample, error)
+	// Sample returns share and corresponding proof for the given axis and share index in this axis.
+	Sample(ctx context.Context, rowIdx, colIdx int) (shwap.Sample, error)
 	// AxisHalf returns Shares for the first half of the axis of the given type and index.
 	AxisHalf(ctx context.Context, axisType rsmt2d.Axis, axisIdx int) (AxisHalf, error)
-	// Data returns data for the given namespace and row index.
-	Data(ctx context.Context, namespace share.Namespace, rowIdx int) (shwap.RowNamespaceData, error)
+	// RowData returns data for the given namespace and row index.
+	RowData(ctx context.Context, namespace share.Namespace, rowIdx int) (shwap.RowNamespaceData, error)
 	// EDS returns extended data square stored in the file.
 	EDS(ctx context.Context) (*rsmt2d.ExtendedDataSquare, error)
 }
```

Comment:
* Most of the comments reference a file, and this is not a file anymore
* RowData -> RowNamespaceData
* The EDS method on the EDS interface makes me feel uneasy. I would rename either of those to something like `Square`, preferably the interface. Then, the EDS method would return the canonical rsmt2d struct.
 * This also made us wonder what the Getter interface should return: the actual EDS or the interface. I remember we had issues with serialization on rsmt2d side, which we could solve on our layer as well.

Reply:
> EDS(ctx context.Context) (*rsmt2d.ExtendedDataSquare, error)
Perhaps only method can be renamed to `ToRSMT2D`?